File: src/main/java/net/infstudio/goki/common/stats/Stat.java
package net.infstudio.goki.common.stats;

import net.minecraft.entity.player.EntityPlayer;

public interface Stat {
    boolean needAffectedByStat(Object... obj);

    float[] getAppliedDescriptionVar(EntityPlayer player);

    float getBonus(int level);

    float getBonus(EntityPlayer player);

//	abstract float getBonus(int paramInt);

    float getAppliedBonus(EntityPlayer player, Object paramObject);

    int getCost(int level);

    int getLimit();

//	public String getSimpleDescriptionString();
}

File: src/main/java/net/infstudio/goki/api/stat/Stat.scala
package net.infstudio.goki.api.stat

import net.minecraft.entity.player.EntityPlayer

trait Stat {

    def needAffectedByStat(obj: Any*): Boolean

    def getAppliedDescriptionVar(player: EntityPlayer): Array[Float]

    def getBonus(level: Int): Float

    def getBonus(player: EntityPlayer): Float

    def getAppliedBonus(player: EntityPlayer, paramObject: Any): Float

    def getCost(level: Int): Int

    def getLimit: Int

    def getKey: String
}
